opentelemetry-sdk/src/metrics/meter.rs

```diff
@@ -533,6 +550,25 @@ fn validate_instrument_config(name: &str, unit: &Option<Cow<'static, str>>) -> M
     validate_instrument_name(name).and_then(|_| validate_instrument_unit(unit))
 }
 
+fn validate_buckets(buckets: &[f64]) -> MetricResult<()> {
+    if buckets.is_empty() {
+        return Err(MetricError::InvalidInstrumentConfiguration(
+            "Buckets must not be empty",
+        ));
+    }
+
+    // validate that buckets are sorted and non-duplicate
+    for window in buckets.windows(2) {
```

Comment:
I think it's more readable (also a bit more efficient) to use a simple `for` loop here.

```rust
if let Some(boundaries) = builder.boundaries {
    for i in 1..boundaries.len() {
        if boundaries[i] <= boundaries[i - 1] {
            return Err(MetricError::InvalidInstrumentConfiguration(
                "Buckets must be sorted and non-duplicate",
            ));
        }
    }
}
```